### PR TITLE
DOV-5408: Document integration of imap_zlib into core

### PR DESCRIPTION
**Pull request by fleischie**
This PR documents the integration - and subsequent removal of - the imap_zlib plugin into the core IMAP functionality:

- Add relevant `versionchanged`/`versionremoved` statements to usage of imap_zlib plugin, and
- add reference of removal into upgrading guide.

Closes DOV-5408.

### CODE REVIEW

File: source/settings/plugin/imap-zlib-plugin.rst [hunk 1]
```diff
@@ -6,6 +6,13 @@ imap-zlib plugin
 
 See :ref:`imap_compress` for an overview of Dovecot's IMAP compression support.
 
+.. versionremoved:: v2.4.0;v3.0.0
+
+The imap-zlib plugin has been integrated into the main imap functionality of
+dovecot. The ``COMPRESS`` IMAP extension is now automatically enabled as long
+as the project has been compiled with the ``zlib`` library, which is
```
Comment (cmouse) [inline source/settings/plugin/imap-zlib-plugin.rst:13]: This is currently required, not optional, in your code merge request.

File: source/installation_guide/upgrading/from-2.3-to-3.0.rst [hunk 2]
```diff
@@ -94,6 +94,9 @@ Removed features and their replacements
 +------------------------------------------------------------+------------------------------------------------------------------------------------------+
 | OpenSSL support for older than 1.0.2                       | Older versions are not supported anymore.                                                |
 +------------------------------------------------------------+------------------------------------------------------------------------------------------+
+| :ref:`plugin-imap-zlib`                                    | The IMAP ``COMPRESS`` extension is now automatically enabled if dovecot has been         |
+|                                                            | compiled with the ``zlib`` library. See :ref:`imap_compress`.                            |
```
Comment (cmouse) [inline source/installation_guide/upgrading/from-2.3-to-3.0.rst:98]: same here as well..

**Review by cmouse — CHANGES_REQUESTED** [review]

File: source/configuration_manual/imap_compress.rst [hunk 3]
```diff
@@ -13,11 +13,32 @@ IMAP session.
...
 
 Configuration
 -------------
 
+To configure the compression level for the used mechanism use the following
+option:
+
+.. code-block:: none
+
+  # syntax: imap_compress_<algorithm>_level = value
```
Comment (cmouse) [inline source/configuration_manual/imap_compress.rst:32]: didn't we drop this?

**Review by cmouse — CHANGES_REQUESTED** [review]

File: source/configuration_manual/imap_compress.rst (hunk 3 above)
Comment (fleischie) [reply source/configuration_manual/imap_compress.rst:32]: Yes. The first block in the settings section is that it was removed in v2.4.0/v3.0.0 There is unfortunately no obvious way to make the rest of the document structured below it without sacrificing readibility.

Comment (cmouse) [reply source/configuration_manual/imap_compress.rst:32]: We could just remove this block completely? It's still going to be available in 2.3 docs.

**Review by cmouse — APPROVED** [review]

**Pull request merged**